app_dart/lib/src/model/luci/build_pubsub.dart

```diff
@@ -0,0 +1,926 @@
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:cocoon_service/src/model/common/json_converters.dart';
+import 'package:cocoon_service/src/request_handling/body.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+part 'build_pubsub.g.dart';
+
```

Comment:
Can you add a doc linking to the schema from LUCI side?

Reply:
Yeah will do.

Reply:
Rethinking here. Does it make sense to consolidate all these to buildbucket.dart?

Reply:
I can but it will make for a single file that is especially long.

Reply:
I also avoided that in the beginning because there are multiple "instances" of the same type which are used throughout the protos that are nested inside other message types. This was making it hard to keep things straight.

Reply:
Huh, do you think we can address the nested types/instances/protos (I would expect not too many) first, and then just add those new classes?
Consolidating those two model files makes sense to considering everything in `build_pubsub.dart` is related to `buildbucket`. Sounds technical debts to keep two files which share a good number of same classes.

Reply:
The nested protos are in that file. That is why the file is so big. There are a few different instances of input depending on where it is nested. So not very reusable. I can put it all in one class though not my preference. Also for some reason when you serialize json with dart all the classes need to be in the same file or it doesn't work.